### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by bxin**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -44,18 +44,589 @@
...
+
+   Document-18499 :cite:`Document-18499` defines a set of CS and the transformations between them. However, it doesn't go into all the levels of details that are sufficient for understanding the AOS operations.
+
+.. _section-ocs:
+
+#############################
+The Optical Coordinate System
+#############################
+
+Our AOS system-wide CS is the Optical Coordinate System (OCS). The defition of the OCS is the same as defined in LTS-136 :cite:`LTS-136`, Section 2.2. In summary -
```
Comment (teweitsai) [inline index.rst]: The "definition" of the OCS

File: index.rst [hunk 2]
```diff
@@ -44,18 +44,589 @@
...
+- M1M3
+- M2
+- M2 hexapod
+- Camera hexapod
+- Camera rotator
+- ComCam (during commissioning)
+- LSSTCam (during commissioning and operations)
+
+Over the years, each component has been under development under their own coordinate system (CS).
+Sometimes the situation can be even more complicated - A single vendor may be using multiple CS for different sub-components (for example, mechanical versus optical). There are also inconsistencies between various engineering models and drawings.
```
Comment (teweitsai) [inline index.rst]: may be using multiple "CSs" for

File: index.rst [hunk 3]
```diff
@@ -44,18 +44,589 @@
...
+The Optical Coordinate System
+#############################
+
+Our AOS system-wide CS is the Optical Coordinate System (OCS). The defition of the OCS is the same as defined in LTS-136 :cite:`LTS-136`, Section 2.2. In summary -
+
+- The origin of the OCS is at the theoretical vertex of M1.
+- +z axis points along the optical axis, toward the sky
+- +x axis goes in parallel with the elevation axis. When look from the sky, +x points to the right.
+- +y axis can then be determined using the right hand rule. At horizon pointing, +y goes up.
+- The azimuth angle is zero when the telescope points toward northern horizon. Looking from the sky, it increase clockwise.
```
Comment (teweitsai) [inline index.rst]: Looking from the sky, it increase"s" clockwise.

File: index.rst [hunk 4]
```diff
@@ -44,18 +44,589 @@
...
+- The zenith angle is zero when the telescope points at zenith. It increase to 90 degrees for horizon pointing.
```
Comment (teweitsai) [inline index.rst]: It increase"s" to 90 degrees

File: index.rst [hunk 5]
```diff
@@ -44,18 +44,589 @@
...
+.. figure:: /_static/ocs.png
+   :name: fig-ocs
+   :target: ../_images/ocs.png
+   :alt: The Optical Coordinate System (OCS)
+
+Optical Coordinate System (OCS). Left: view from the sky; Right: side view from the -X axis.
+
+The idea is that the OCS is fixed to M1, which defines the reference for the optical system.
+Relative to the ground, the OCS rotates with the azimuth angle and the zenith angle.
+An easy way to identify the +x/+y/+z while looking at a drawing or the actual hardware is that, the walkway on the TMA, above M1M3, is in the +y direction. Otherwise when we go toward horizon it is going to bump into the ground.
```
Comment (teweitsai) [inline index.rst]: I do not understand this sentence at all. Please help to explain more. Thanks!

File: index.rst [hunk 6]
```diff
@@ -44,18 +44,589 @@
...
+
+.. note::
+   The Telescope and Site top-level CS document is LTS-136 :cite:`LTS-136`, and the Camera top-level CS document is LCA-280 :cite:`LCA-280`. Unlike those two documents, we do not define the Observatory Mount CS or the Azimuth CS, because these CS, as defined by LTS-136 :cite:`LTS-136` and LCA-280 :cite:`LCA-280`, actually contradict each other, in terms of which axis points north, which points west etc. We also avoid the use of terms such as ``elevation angle`` and ``altitude``. Instead, we always use ``zenith angle``, as defined above.
```
Comment (teweitsai) [inline index.rst]: which points west"," etc.

File: index.rst [hunk 7]
```diff
@@ -44,18 +44,589 @@
...
+
+
+################
+Zemax and PhoSim
+################
+
+We discuss the Zemax and PhoSim CS first, because these are relatively easier to define -
```
Comment (teweitsai) [inline index.rst]: Zemax and PhoSim CS"s"

File: index.rst [hunk 8]
```diff
@@ -44,18 +44,589 @@
...
+they are self-consistent within their own framework.
+When we say Zemax CS, we refer to the global CS as used by
```
Comment (teweitsai) [inline index.rst]: Which "global CS" you are talking about here:
https://confluence.lsstcorp.org/pages/viewpage.action?spaceKey=SYSENG&title=As-built+optical+model

File: index.rst [hunk 9]
```diff
@@ -44,18 +44,589 @@
...
+
+.. code-block:: py
+
+   def zcs2ocs(x,y,z):
+       return -x,y,-z
+   def ocs2zcs(x,y,z):
+       return -x,y,-z
+
+The optical sensitivity matrix (senM) is derived using the Zemax optical model.
+Therefore, everything about the senM follows the ZCS. We were able to close the simulation loop with PhoSim, because we made PhoSim consistent with Zemax.
```
Comment (teweitsai) [inline index.rst:147]: Is ZCS = ZEMAX CS?

File: index.rst [hunk 10]
```diff
@@ -44,18 +44,589 @@
...
+With the actual hardware, we will need to convert all commands returned the AOS control into the proper CS of each component before they are applied.
+
+.. note::
+
+    Note that we apply the decenters and tilts in Zemax via ``Coordinate Breaks``. Mathematically the order of decenters and tilts matter. In Zemax, there is a ``order flag``. When it is set to 0, Zemax does the decenters first, then x-tilt, y-tilt, z-rotation. When the ``order flag`` is set to 1, Zemax does these in exact opposite order, so that users can easily go back to the original CS :cite:`Zemax13manual`. However, in the AOS context, we don't really care about these because the tilts are always small enough for the order not to make a difference. If this is not true, then the basic approach of taking the decenters and tilts of the hexapods as independent variable in the AOS control wouldn't be correct.
```
Comment (teweitsai) [inline index.rst]: When you say the tilts are small enough, can you specify how small it should be to fulfill this?

File: index.rst [hunk 11]
```diff
@@ -44,18 +44,589 @@
...
+
+####
+M1M3
+####
+
+The M1M3 glass mirror was casted and polished at the University of Arizona Richard F. Caris Mirror Lab (RFCML).
+The mirror cell was made by CAID Industriess, and software is designed and written by the Rubin Obs. team.
```
Comment (teweitsai) [inline index.rst]: Industries??? I am not sure this is typo or not.

File: index.rst [hunk 12]
```diff
@@ -44,18 +44,589 @@
...
+
+Like other components of the AOS, M1M3 operates mostely off its Look-Up Table (LUT), which contains our best knowledge of the forces as functions as gravity (or zenith angle) and temperature profiles on and around the mirror surfaces. The current M1M3 LUT can be found `here <https://github.com/lsst-sitcom/M1M3_ML/blob/master/data/FLUT.yaml>`__.
+
+- The zenith angle, as the primary input to the M1M3 LUT, is defined the same way as the OCS zenith angle as defined in Sec. :ref:`section-ocs`.
+- Unrelated to the bending modes, but relevant to the LUT, are the forces on the secondary cylinders of the lateral and bilateral actuators. When the force on the secondary cylinder of an lateral actuator is positive, it pushes M1M3 in the y-z plane, along 45 degrees between +y and +z axes. When the force on the secondar cylinder of a bilateral actuator is positive, it pushes M1M3 in the x-z plane, along the 45 degree line between either the +z and +x or the +z and -x directions, depending on the location of the bilateral actuator.
```
Comment (teweitsai) [inline index.rst]: Can you specify which bilateral actuators do "+z and +x" or "+z and -x" in M1M3?

File: index.rst [hunk 13]
```diff
@@ -44,18 +44,589 @@
...
+       '''
+       return x,y,z-d_L1_M1
+   def ccs2ocs(x,y,z, d_L1_M1):
+       return x,y,z+d_L1_M1
+   def dvcs2ccs(x,y,z):
+       return y,x,z
+   def ccs2dvcs(x,y,z):
+       return y,x,z
+
+The wavefront sensors are rotated on the focal plane. The wavefront sensor images we get from the DAQ will need to be rotated to be put into the CCS. See `here in the IM code <https://github.com/bxin/IM/blob/9d74b83eb15021e91d27bf96aa262ff378550818/source/aosWFS.py#L402>`__ or `here in ts_wep code <https://github.com/lsst-ts/ts_wep/blob/master/python/lsst/ts/wep/WepController.py#L408-L421>`__.
```
Comment (teweitsai) [inline index.rst]: There was the discussion to unify the coordinate systems of corner WFS before (in the software CCS or DM side). If this was the case in the final, we would not need to do this rotation anymore. The decision and related work are still TBD at this moment.

File: index.rst (hunk 8 above)
Comment (bxin) [reply index.rst]: The confluence page links to zmx file. You will need to open the zmx file to see the CS it uses.

File: index.rst (hunk 10 above)
Comment (bxin) [reply index.rst]: well, it depends on how accurate you want it to be, i.e., what your tolerance is on the difference due to the ordering. I guess one can do some simple numerical experiments to get an idea (I played with it years ago). Or, the fact that we take them as independent variables and still can make AOS converge is a proof that the ordering does not matter for practical purposes.

File: index.rst (hunk 8 above)
Comment (teweitsai) [reply index.rst]: Not everyone contains me has the ZEMAX license. Is it possible to explain it in the document directly? Thanks!

Comment (bxin) [reply index.rst]: The document already defines ZCS.

File: index.rst (hunk 5 above)
Comment (bxin) [reply index.rst]: will add annotation to the figure to point out walkway.

File: index.rst (hunk 13 above)
Comment (bxin) [reply index.rst]: will add a note. thanks.

**Review by sandrinethomas — COMMENTED** [review]
This document looks good and will be very useful during commissioning!
Here are my comments (very little)
- Even though you define what OCS is in the context of this document, I think it would be better to go away from the software OCS acronym. What about OpCS?
- In the M1M3 section, it might be good to add a drawing for the bending modes coordinate systems
- Same comment for M2
- For the alignment system, there might be some rotation to take into account as well, unless we always set the rotator to 0 while checking the alignment (which is feasible)

**Comment by bxin** [comment]
Sandrine - 
Thanks for the comments. 
1. I think there should not be much confusion since the other OCS is in the software context. And I don’t want to create yet another new acronym. So I prefer to use OCS here, unless you have strong objections 🙂
2. Good idea. I added links to the bending mode plots. I didn’t include those plots in this note because this note is not really about bending modes, so I didn’t want those plots to take up too much space.
3. Same as 2.
4. I rewrote the ASC section a bit….

File: index.rst [hunk 14]
```diff
@@ -609,7 +612,13 @@ For example, in the Zemax model,
...
+The way the Alignment System Control (ASC) works is to measure the rigid body positions of M2 and the camera in the M1M3 CS or some CSs that are tied to the M1M3 CS, determine the offset relative to their reference positions, then command the hexapods to move M2 and the camera to their reference positions.
+
+To facilitate the above operations, it would be the easiest to have the ASC report target positions in the M2 CS and CCS, for M2 and the camera, respectively, so that the inverse of the offsets can be sent to the hexapods without further coordinate transformations.
+After a command is issued to measure the position of a target (M2 or Camera), a measurement plan is executed by the Spatial Analyzer (SA), which measures the x,y, and z of all the Spherically Mounted Reflectors (SMSs), and fit them to an internal model of the target. The CSs used by the SA are configurable.
```
Comment (teweitsai) [inline index.rst]: "x,y, and z" --> "x, y, and z".

**Pull request merged**